ScoutSuite/core/server.py

```diff
@@ -1,176 +1,18 @@
-        :return:                        The server object.
-        """
-        self.results = SqliteDict(filename)
-
-    @cherrypy.expose()
-    @cherrypy.tools.json_out()
-    def summary(self):
-        """
-        Returns the stripped down data of the results that doesn't scale up when using a lot of resources,
-        used to render the summary.
-        Should be the first call from the server.
-        Can be found at GET /api/summary
-
-        :return:                        The summary data of the report.
-        """
-        data = dict(self.results)
-        services = data.get('services')
-        stripped_services = {}
-        for k1, v1 in services.items():
-            service = {}
-            for k2, v2 in v1.items():
-                if k2 == 'findings' or k2 == 'filters' or count_re.match(k2):
-                    service[k2] = v2
-            stripped_services[k1] = service
-        data['services'] = stripped_services
-        return {'data': data}
-
-    @cherrypy.expose()
-    @cherrypy.tools.json_out()
-    def data(self, key=None):
-        """
-        Return the data at the requested key. Doesn't returns nested dictionaries and lists.
-        If one of the value is a dictionary, it will return {'type': 'dict', 'keys': <Array of all the keys>}
-        If one of the value is a list, it will return {'type': 'list', 'count': <number of elements in the list>}
-
-        Can be found at GET /api/data?key=<KEY>
-        :param key:                     Key of the requested information, separated by the character '¤'.
-        :return:                        The data at the requested location stripped of its nested data.
-        """
-        result = self.get_item(self.results, key)
-        # Returns only indexes or length if it's a complex type
-        if isinstance(result, dict) or isinstance(result, SqliteDict):
-            result = {'type': 'dict', 'keys': list(result.keys())}
-        elif isinstance(result, list):
-            result = {'type': 'list', 'length': len(result)}
-        return {'data': result}
-
-    @cherrypy.expose()
-    @cherrypy.tools.json_out()
-    def full(self, key=None):
-        """
-        Return the data at the requested key. Returns all the nested data.
-        Be sure not to use it on a key that may contains a lot of data, as the request won't be answered
-        if it's too large(generally 3mb).
-
-        Can be found at GET /api/full?key=<KEY>
-        :param key:                     Key of the requested information, separated by the character '¤'.
-        :return:                        The data at the requested location.
-        """
-        result = self.get_item(self.results, key)
-        if isinstance(result, str) or isinstance(result, int):
-            return {'data': result}
-        return {'data': dict(result)}
-
-    @cherrypy.expose()
-    @cherrypy.tools.json_out()
-    def page(self, key=None, page=None, pagesize=None):
```

Comment:
@spoilist looks like this implements pagination, maybe it could be of use to you / give you some ideas?